modules/ardrivo/src/Wire.cxx

```diff
@@ -0,0 +1,140 @@
+{
+    //Begin to the specific "adress"???
+    begun = true;
+}
+void TwoWire::begin(int adress)
+{
+    //Begin to the specific "adress"???
+
+    begun = true;
+}
+
+size_t TwoWire::requestFrom(uint8_t address, size_t size, bool stop)
+{
+    // ?????
+    // How do you diffrentiate REQUEST_SEND och REQUEST_RECEIVE????
+    /*    
+    REQUEST_SEND = 1,
+    REQUEST_RECEIVE
+    */
+
+    //vad faeeen g�r x??
+    //int x = 0 och x i ic2_buses[x] ????
+
+    //med denna gn�ller den inte
+    ProtocolRequest req = ProtocolRequest(RequestType::REQUEST_SEND, size);
+    transmissionAdress = address;
+    
+    //med denna gn�ller den 
+    //auto req = new ProtocolRequest(RequestType::REQUEST_SEND, size);
+    board_data->i2c_buses[bus_id].requests.at(address).push_back(req);
+
+    // need other return value
+    return 0;
+}
+
+uint8_t TwoWire::requestFrom(uint8_t address, uint8_t size)
+{
+    return requestFrom((uint8_t)address, (uint8_t)size, (uint8_t)true);
+}
+uint8_t TwoWire::requestFrom(uint8_t address, uint8_t size, uint8_t stop)
+{
+    return requestFrom((uint8_t)address, (uint8_t)size, (uint8_t)true);
+}
+uint8_t TwoWire::requestFrom(int address, int size)
+{
+    return requestFrom((uint8_t)address, (uint8_t)size, (uint8_t)true);
+}
+uint8_t TwoWire::requestFrom(int address, int size, int stop)
+{
+    return requestFrom((uint8_t)address, (uint8_t)size, (uint8_t)stop);
+
+}
+
+void TwoWire::beginTransmission(uint8_t adress)
+{
+    transmitting = true;
+    // set address of targeted slave    
+    transmissionAdress = adress;
+}
+void TwoWire::beginTransmission(int adress)
+{
+    beginTransmission((uint8_t)(adress));  
+}
+
+uint8_t TwoWire::endTransmission()
+{
+    return endTransmission(true); 
+}
+uint8_t TwoWire::endTransmission(uint8_t)
+{
+    if (transmitted > 0)
+    {
+        transmitting = false;
+        //0 = sucess transmission
+        return 0;
+    }
+    transmitting = false;
+    //Error at transmission
+    return 1;
+};
+
+size_t TwoWire::write(uint8_t amount)
+{
+    if (transmitting)
+    {
+        // in master transmitter mode
+        // don't bother if buffer is full
+        //something to care about???
+        //if(txBufferLength >= BUFFER_LENGTH)
+
+        //else
+        // put byte in tx buffer
+        board_data->i2c_buses[transmissionAdress].tx.fill((std::byte)amount); //Would this work?
+
+        transmitted = amount;
+        //update amount in buffer???
+        return transmitted;
+    }   
+    return 0;
+}
+
+size_t write(const uint8_t *, size_t); // do what??
+
+
+int TwoWire::available(void)
+{
+    //get avaiable bytes from what device??
+    //This should be called on a master device 
+
+    std::byte stuff = board_data->i2c_buses[/*what adress??*/].tx.at(transmissionAdress);
+    return (int)stuff;
+}
+int TwoWire::read(void)
+{
+    int data;
+    //weird as error stuff with bus_id
+ 
+    data = board_data->i2c_buses[/*what adress??*/ bus_id].rx.front;
```

Comment:
You raise an important question; I'd say latest on the bus, but you should do some research on I2C to see if a device discards its previous buffer or not when written to.

This also raises up the fact that `i2c_buses` should be a vector of hashmaps, where the vector index maps the I2C config index
Keep in mind that there can be multiple I2C endpoints on a single board, that could each be on a different bus (not sure how we would solve the problem of potentially having the board talk to itself by putting both a master and a slave endpoint on the same bus)